src/numberGenerator.js

```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+const getFourDigitNumber = () => {
```

Comment:
For now, that's the hardest implementation I've seen for this function related to this task. The second important thing of code after working properly, it's to be friendly to other developers, and these lines are super aggressive. To check it, try to ask some students you have good communication with to tell you what you are trying to do there without showing them the name of the function. Think about how you can simplify it.